### DISCUSSION TITLE
How to view the dataset after data augmentation

### DISCUSSION THREAD

**Discussion by casekevin** [MIC-DKFZ/nnUNet#1114] [Q&A] [answered]
Hello 

I am curious about how nnUNet decides the augmentation to the dataset, so is there ways we could extract the dataset after the data augmentation before the initialization of the network?

Thank you!

**Comment by FabianIsensee** [comment, marked as answer]
data augmentation is done on the fly during training, so there is no augmented dataset you can look at. That you can do is save the samples that the network sees in run_iteration

**Reply by casekevin** [reply]
Thank you so much!

**Reply by jmrichusc** [reply]
Hello,

Could you explain how to save the samples that the network sees in run_iteration for nnUNetTrainerV2?

Also, how would one view the augmented dataset in nnUNetTrainerV2_DA5?

**Reply by FabianIsensee** [reply]
You need to code this yourself. You have already identified the right location. Just add some code that exports the batches
